clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -741,12 +1057,88 @@ impl<'a, 'b> TypeChecker<'a, 'b> {
         }
     }
 
-    fn inner_type_check(
+    fn clarity1_type_check_expects(
+        &mut self,
+        expr: &SymbolicExpression,
+        context: &TypingContext,
+        expected_type: &TypeSignature,
+    ) -> TypeResult {
+        match (&expr.expr, expected_type) {
+            (
+                LiteralValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))),
+                TypeSignature::CallableType(CallableSubtype::Trait(trait_identifier)),
+            ) => {
+                runtime_cost(
+                    ClarityCostFunction::AnalysisFetchContractEntry,
+                    &mut self.cost_track,
+                    1,
```

Comment:
Shouldn't the cost here be equal to the size of the fetched contract code?

Reply:
Right, I guess I can't improve on the cost checking for the Clarity1 paths. I will change this back to match the old implementation.

Reply:
Fixed in 7273dd01e

Reply:
In general, all Clarity 1 code paths should continue to use the same cost assessment since it's consensus-critical code.  I'm not sure if there are others.